### PR TITLE
check and update two slots

### PR DESCRIPTION
**Pull request by livingrockrises**

### CODE REVIEW

File: contracts/Nexus.sol [hunk 1]
```diff
@@ -285,9 +299,20 @@ contract Nexus is INexus, EIP712, BaseAccount, ExecutionHelper, ModuleManager, U
     }
 
     /// Upgrades the contract to a new implementation and calls a function on the new contract.
+    /// @notice Updates two slots 1. 1967 slot and 2. address() slot in case if it's upgraded earlier from Biconomy V2 account.
```
Comment (filmakarov) [inline contracts/Nexus.sol:302]: ...as Biconomy v2 Account (proxy) reads implementation from the slot that is defined by its address.

**Review by filmakarov — APPROVED** [review]

File: contracts/Nexus.sol [hunk 2]
```diff
@@ -237,6 +237,20 @@ contract Nexus is INexus, EIP712, BaseAccount, ExecutionHelper, ModuleManager, U
...
+    /// @notice Retrieves the address of the current implementation from the EIP-1967 slot.
+    /// @notice Checks the 1967 implementation slot, if not found then checks the slot defined by address (Biconomy V2 smart account)
+    /// @return implementation The address of the current contract implementation.
+    function getImplementation() external view returns (address implementation) {
+        assembly {
+            implementation := sload(_ERC1967_IMPLEMENTATION_SLOT)
+        }
+        if(implementation == address(0)) {
+            assembly {
+            implementation := sload(address())
```
Comment (Aboudjem) [inline contracts/Nexus.sol:249]: what is this magic?

Comment (livingrockrises) [reply contracts/Nexus.sol:249]: which line?

Comment (filmakarov) [reply contracts/Nexus.sol:249]: `address()` opcode returns the same as `address(this)` 
so the contract's address is used as number of the storage slot (it is converted to bytes32) to store the address of the implementation

so this line loads it from storage using the opcode to obtain the slot where the implementation address is stored.

Comment (livingrockrises) [reply contracts/Nexus.sol:249]: yup,
https://github.com/bcnmy/scw-contracts/blob/main/contracts/smart-account/Proxy.sol#L23

File: contracts/Nexus.sol (hunk 1 above)
Comment (livingrockrises) [reply contracts/Nexus.sol:302]: fixed

**Pull request merged**